analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/InvalidCastToInterface.cs

```diff
@@ -0,0 +1,164 @@
+﻿using System;
+using System.Collections.Generic;
+
+public interface IMyInterface { }
+public interface IMyInterface2 { }
+public interface IMyInterface3 : IMyInterface { }
+public interface IMyInterface4 : IMyInterface { }
+public interface IMyInterface5 { }
+
+public class Implementer : IMyInterface { }
+public class Implementer5 : IMyInterface5 { }
+
+public class MyClass1 { }
+public class MyClass2 { }
+public class MyClass3 : MyClass2, IMyInterface { }
+public class MyClass4 { }
```

Comment:
The naming here is horribly arbitrary, which makes understanding the tests very hard.